code/modules/antagonists/changeling/powers/biodegrade.dm

```diff
@@ -91,3 +102,60 @@
 	if(C && user.loc == C)
 		qdel(C) //The cocoon's destroy will move the changeling outside of it without interference
 		to_chat(user, "<span class='warning'>We dissolve the cocoon!</span>")
+
+/datum/action/changeling/biodegrade/proc/add_hand_spell(mob/living/carbon/human/user)
+	if(user.get_active_hand() && !user.drop_item())
+		to_chat(user, "<span class='warning'>[user.get_active_hand()] is stuck to our hand, we cannot emit acid on this hand.</span>")
+		return FALSE
+	current_hand = new hand(src)
+	user.put_in_active_hand(current_hand)
+	RegisterSignal(user, COMSIG_MOB_WILLINGLY_DROP, PROC_REF(remove_hand_spell))
+	return TRUE
+
+/datum/action/changeling/biodegrade/proc/remove_hand_spell(mob/living/carbon/human/user, any_hand=FALSE)
+	SIGNAL_HANDLER
+	if(!current_hand)
+		return FALSE
+	if(any_hand && user.get_active_hand() != current_hand)
+		return FALSE
+	qdel(current_hand)
+	UnregisterSignal(user, COMSIG_MOB_WILLINGLY_DROP)
+	return TRUE
+
+/obj/item/melee/changeling_corrosive_acid
+	name = "Corrosive acid"
+	desc = "A fistfull of death."
+	icon_state = "alien_acid"
+	item_state = null
+	flags = ABSTRACT | NODROP | DROPDEL
+	w_class = WEIGHT_CLASS_HUGE
+	force = 0
+	throwforce = 0
+	throw_range = 0
+	throw_speed = 0
+	var/datum/action/changeling/biodegrade/parent_action
+
+/obj/item/melee/changeling_corrosive_acid/New(datum/action/changeling/biodegrade/new_parent)
+	. = ..()
+	parent_action = new_parent
+
+/obj/item/melee/changeling_corrosive_acid/afterattack(atom/target, mob/user, proximity, params)
+	if(target == user)
+		to_chat(user, "<span class='noticealien'>You withdraw your readied acid.</span>")
+		parent_action.remove_hand_spell(user)
+		return
+	if(isliving(target))
+		to_chat(user, "<span class='alert'>We dont have enough acid to assault living target!</span>")
+		return
+	if(!proximity || !iscarbon(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED)) // Don't want xenos ditching out of cuffs
+		return
+	if(target.acid_act(200, 100))
+		visible_message("<span class='alert'>[user] vomits globs of vile stuff all over [target]. It begins to sizzle and melt under the bubbling mess of acid!</span>")
+		add_attack_logs(user, target, "Applied corrosive acid") // Want this logged
+	else
+		to_chat(user, "<span class='noticealien'>You cannot dissolve this object.</span>")
+	parent_action.remove_hand_spell(user)
+
+/obj/item/melee/changeling_corrosive_acid/Destroy()
+	parent_action.current_hand = null
+	return ..()
```

Comment:
please move this to directly under the `New()` proc

Reply:
sorry what ya mean? Setting `current_hand` in `New` proc instead of `/datum/action/changeling/biodegrade/proc/add_hand_spell`?

Reply:
as in, the entire proc definition should be moved

```dm
/type/definition
	...

/type/New()
	...

/type/Destroy()
	...

/type/all_other_procs()
	...
```

Reply:
moved to this order
also moved UnregisterSignal to Destroy to prevent getting runtimes whenever hand item is deleted without spell proc and user wants to activate new hand